front/src/modules/timetableItem/components/Timetable/RoundTrips/RoundTripsModal.tsx

```diff
@@ -118,12 +161,65 @@ const RoundTripsModal = ({
     setRoundTripsModalIsOpen(false);
   };
 
+  const saveRoundTrips = async () => {
+    const trainScheduleIdsToDelete: number[] = [];
+    const trainScheduleOneWaysIds: number[] = [];
+    const trainScheduleRoundTripsIds: number[][] = [];
+    const pacedTrainIdsToDelete: number[] = [];
+    const pacedTrainOneWaysIds: number[] = [];
+    const pacedTrainRoundTripsIds: number[][] = [];
+
+    for (const item of pairingItems) {
+      if (isTrainScheduleId(item.id) && trainScheduleRoundtrips) {
+        const itemRawId = extractEditoastIdFromTrainScheduleId(item.id);
+        buildRoundTripsPayload({
+          item,
+          itemRawId,
+          roundTrips: trainScheduleRoundtrips,
+          oneWaysIds: trainScheduleOneWaysIds,
+          roundTripsIds: trainScheduleRoundTripsIds,
+          deleteIds: trainScheduleIdsToDelete,
+        });
+      }
+      if (isPacedTrainId(item.id) && pacedTrainRoundtrips) {
+        const itemRawId = extractEditoastIdFromPacedTrainId(item.id);
+        buildRoundTripsPayload({
+          item,
+          itemRawId,
+          roundTrips: pacedTrainRoundtrips,
+          oneWaysIds: pacedTrainOneWaysIds,
+          roundTripsIds: pacedTrainRoundTripsIds,
+          deleteIds: pacedTrainIdsToDelete,
+        });
+      }
+    }
```

Comment:
Nit: maybe it would be better to handle this in a utils function to make saveRoundTrips lighter (but you will have to adapt `buildRoundTripsPayload`
```suggestion
    if (!dbTrainScheduleRoundtrips || !dbPacedTrainRoundtrips) return;

    const {
      trainScheduleOneWaysIds,
      trainScheduleRoundTripsIds,
      trainScheduleIdsToDelete,
      pacedTrainOneWaysIds,
      pacedTrainRoundTripsIds,
      pacedTrainIdsToDelete,
    } = buildRoundTripsPayload(pairingItems, dbTrainScheduleRoundtrips, dbPacedTrainRoundtrips);
```

Reply:
Done